phpdotnet/phd/Package/Generic/XHTML.php

```diff
@@ -2039,6 +2045,110 @@ public function format_itemizedlist($open, $name, $attrs, $props) {
+            }
+
+            return '<ul class="' . $name . '">';
+        }
+
+        if ($this->cchunk["simplelist"]["type"] === "inline") {
+            $list = "";
+            foreach ($this->cchunk["simplelist"]["members"] as $member) {
+                $list .= $member . ", ";
+            }
+            $list = rtrim($list, ", ");
+
+            $this->cchunk["simplelist"] = $this->dchunk["simplelist"];
+            return $list . '</span>';
+
+        }
+
+        if ($this->cchunk["simplelist"]["type"] === "horiz") {
+
+            $table = "";
+            for ($i = 0; $i < count($this->cchunk["simplelist"]["members"]); $i++) {
+                if ($i % $this->cchunk["simplelist"]["columns"] === 0) {
+                    $table .= str_repeat(" ", $props["depth"] + 2) . "<tr>\n";
+                }
+
+                $table .= str_repeat(" ", $props["depth"] + 3) . "<td>" . $this->cchunk["simplelist"]["members"][$i] . "</td>\n";
+
+                if ($i % $this->cchunk["simplelist"]["columns"] === $this->cchunk["simplelist"]["columns"] - 1) {
+                    $table .= str_repeat(" ", $props["depth"] + 2) . "</tr>\n";
+                }
+            }
+            if ($i % $this->cchunk["simplelist"]["columns"] !== 0) {
+                $numOfMissingCells = $this->cchunk["simplelist"]["columns"] - ($i % $this->cchunk["simplelist"]["columns"]);
+                $oneRow = str_repeat(" ", $props["depth"] + 3) . "<td></td>\n";
+                $table .= str_repeat($oneRow, $numOfMissingCells);
+                $table .= str_repeat(" ", $props["depth"] + 2) . "</tr>\n";
+            }
+
+            $this->cchunk["simplelist"] = $this->dchunk["simplelist"];
+
+            return $table . str_repeat(" ", $props["depth"] + 1) . "</tbody>\n" . str_repeat(" ", $props["depth"]) . "</table>";
+
+        }
+
+        if ($this->cchunk["simplelist"]["type"] === "vert") {
+
+            $table = "";
+            $numOfRows = ceil(count($this->cchunk["simplelist"]["members"]) / $this->cchunk["simplelist"]["columns"]);
+            for ($row = 0; $row < $numOfRows; $row++) {
+                $table .= str_repeat(" ", $props["depth"] + 2) . "<tr>\n";
+                for ($col = 0; $col < $this->cchunk["simplelist"]["columns"]; $col++) {
+                    $memberIndex = ($numOfRows * $col) + $row;
+                    $table .= str_repeat(" ", $props["depth"] + 3) . "<td>";
+                    if ($memberIndex < count($this->cchunk["simplelist"]["members"])) {
+                        $table .= $this->cchunk["simplelist"]["members"][$memberIndex];
+                    }
+                    $table .= "</td>\n";
+                }
+                $table .= str_repeat(" ", $props["depth"] + 2) . "</tr>\n";
+            }
+            $this->cchunk["simplelist"] = $this->dchunk["simplelist"];
+
+            return $table . str_repeat(" ", $props["depth"] + 1) . "</tbody>\n" . str_repeat(" ", $props["depth"]) . "</table>";
+        }
```

Comment:
I had grouped most of the logic for horizontal and vertical when I was writing a new DocBook renderer: https://gitlab.com/Girgias/docbook-renderer/-/blob/master/src/Tags/SimpleListTagRenderer.php?ref_type=heads

Not sure if anything from there is applicable tho.

Reply:
Ooof. This looks great. I'm dumping all that non-sense I wrote and using your code here, if you're OK with that. I may be adding a couple of comments to make it easier for people like me to understand it though. :-)

Reply:
Feel free to use it and add comments because it is somewhat arcane magic :D

Reply:
It sure seemed like that. :-)

I added some padding around the tags, so don't be surprised when you see that. :-) 

I'll rebase this PR when my other PR is merged and then I only have to fix up a few tests that had their output changed by all these recent PRs. 

At that point, could the tests could be added to the pull request checks? I see that these checks are using PHP 7.1 so that might need a bump to PHP 8.0+ to support parameter and return type type-hinting, match expressions, etc. and `run-test.php` would need to be pulled from `php-src` on each run as well.

Reply:
I've rebased this PR. It should be ready for review again.